packages/shopify-app-express/src/middlewares/ensure-installed-on-shop.ts

```diff
@@ -97,3 +87,78 @@ export function deleteAppInstallationHandler(
     await appInstallations.delete(shop);
   };
 }
+
+function getRequestShop(
+  api: Shopify,
+  config: AppConfigInterface,
+  req: Request,
+  res: Response,
+): string | undefined {
+  if (typeof req.query.shop !== 'string') {
+    config.logger.error(
+      'ensureInstalledOnShop did not receive a shop query argument',
+      {shop: req.query.shop},
+    );
+
+    res.status(500);
```

Comment:
this should be a `422`: https://stackoverflow.com/questions/3050518/what-http-status-response-code-should-i-use-if-the-request-is-missing-a-required
```suggestion
    res.status(422);
```